sourcecode/Part1/command.c

```diff
@@ -0,0 +1,89 @@
+#include "command.h"
+
+void initialiseCommand(Command *command) {
+    command->com_pathname_ = "";
+    command->argc_ = 0;
+    command->redirect_in_ = NULL;
+    command->redirect_out_ = NULL;
+    command->com_suffix_ = NULL;
+    for (int i = 0; i < sizeof(command->argv_) / sizeof(command->argv_[0]); ++i) {
```

Comment:
narrowing conversion here might be dangerous because we're dealing with string sizes, I've refactored and tested this, no warning thrown for my fix:
```c
unsigned long int argv_size = sizeof(command->argv_) / sizeof(command->argv_[0]);  //cache the size calculation.

for (unsigned long int i = 0; i < argv_size; ++i) {
    command->argv_[i] = NULL;
}
```